ts/test/session/sending/MessageQueue_test.ts

```diff
@@ -0,0 +1,393 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import { GroupUtils } from '../../../session/utils';
+import { Stubs, TestUtils } from '../../../test/test-utils';
+import { MessageQueue } from '../../../session/sending/MessageQueue';
+import {
+  ChatMessage,
+  ClosedGroupMessage,
+  ContentMessage,
+  OpenGroupMessage,
+} from '../../../session/messages/outgoing';
+import { PubKey, RawMessage } from '../../../session/types';
+import { UserUtil } from '../../../util';
+import { MessageSender, PendingMessageCache } from '../../../session/sending';
+import { toRawMessage } from '../../../session/utils/Messages';
+import { SessionProtocol } from '../../../session/protocols';
+
+// Equivalent to Data.StorageItem
+interface StorageItem {
+  id: string;
+  value: any;
+}
+
+describe('MessageQueue', () => {
+  // Initialize new stubbed cache
+  let data: StorageItem;
+  const sandbox = sinon.createSandbox();
+  const ourDevice = TestUtils.generateFakePubkey();
+  const ourNumber = ourDevice.key;
+  const pairedDevices = TestUtils.generateMemberList(2).map(m => m.key);
+
+  // Initialize new stubbed queue
+  let messageQueueStub: MessageQueue;
+
+  // Spies
+  let sendMessageToDevicesSpy: sinon.SinonSpy<
+    [Array<PubKey>, ContentMessage],
+    Promise<Array<void>>
+  >;
+  let sendSyncMessageSpy: sinon.SinonSpy<
+    [ContentMessage, Array<PubKey>],
+    Promise<Array<void>>
+  >;
+  let sendToGroupSpy: sinon.SinonSpy<
+    [ContentMessage | OpenGroupMessage],
+    Promise<boolean>
+  >;
+
+  // Message Sender Stubs
+  let sendStub: sinon.SinonStub<[RawMessage, (number | undefined)?]>;
+  let sendToOpenGroupStub: sinon.SinonStub<[OpenGroupMessage]>;
+  // Group Utils Stubs
+  let groupMembersStub: sinon.SinonStub;
+  // Session Protocol Stubs
+  let hasSessionStub: sinon.SinonStub<[PubKey]>;
+  let sendSessionRequestIfNeededStub: sinon.SinonStub<[PubKey], Promise<void>>;
+
+  // Helper function returns a promise that resolves after all other promise mocks,
+  // even if they are chained like Promise.resolve().then(...)
+  // Technically: this is designed to resolve on the next macrotask
+  async function tick() {
+    return new Promise(resolve => {
+      // tslint:disable-next-line: no-string-based-set-timeout
+      setTimeout(resolve, 0);
+    });
+  }
+
+  beforeEach(async () => {
+    // Stub out methods which touch the database
+    const storageID = 'pendingMessages';
+    data = {
+      id: storageID,
+      value: '[]',
+    };
+
+    // Pending Message Cache Data Stubs
+    TestUtils.stubData('getItemById')
+      .withArgs('pendingMessages')
+      .callsFake(async () => {
+        return data;
+      });
```

Comment:
I think this is the same:
```suggestion
      .resolves(data)
```